packages/react-vapor/src/components/limit/Limit.tsx

```diff
@@ -0,0 +1,210 @@
+import classNames from 'classnames';
+import * as VaporSVG from 'coveo-styleguide';
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {InputConnected, Svg} from 'react-vapor';
+import {IReactVaporState} from '../../ReactVapor';
+import {IDispatch} from '../../utils/ReduxUtils';
+import {changeInputValue} from '../input/InputActions';
+import {InputSelectors} from '../input/InputSelectors';
+
+export interface LimitOwnProps {
+    id?: string;
+    title: string;
+    limitLabel?: string;
+    usage?: number;
+    limit?: number;
+    isLimitEditable?: boolean;
+    isLimitTheGoalToReach?: boolean;
+    isHistoryIncluded?: boolean;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: LimitOwnProps) => {
+    const ownLimit: string = ownProps.limit.toString() || Limit.defaultProps.limit.toString();
```

Comment:
Not sure I understand why we are converting the limit to a `string`, because later in the component we are reconverting it to a number with the `+` sign. Wouldn't it be simpler to convert the return value of `InputSelectors.getValue` to a number instead?

Reply:
good point

Reply:
Nice catch! I didn't realize that I could only do one conversion: convert the input (return from the Input Selector), to a number. Doing so, I don't have to convert the limit props, or the currentLimit on the ProgressBar component.